Add documentation for 'setup.cfg' configuration

stephenfin opened the pull request:
An issue with `[egg_info]` was seen with https://github.com/stephenfin/rst2txt. Take the opportunity to highlight this issue along with general `setup.cfg` support.

stephenfin committed:
README: Trivial markup fixes

Consistently mark literals as literals, fix some indentations and
generally clean some things up.

Signed-off-by: Stephen Finucane <stephen@that.guru>

stephenfin committed:
README: Add setup.cfg usage example

This also highlights issues with the '[egg_info]' configuration section
in 'setup.cfg'.

Signed-off-by: Stephen Finucane <stephen@that.guru>

RonnyPfannschmidt reviewed (changes requested):
great job, the markup enhancements are neat and the added informations realy drive home a more metadata oriented approach

for it to be fully embracing of the setuptools features we also need to show people the patterns needed for minimal setuptools versions, else they are in for bad surprises on enterprise distro style setups

stephenfin commented:
> for it to be fully embracing of the setuptools features we also need to show people the patterns needed for minimal setuptools versions, else they are in for bad surprises on enterprise distro style setups

Good call. Done.

stephenfin committed:
README: Add references to pyproject.toml

This is required if you wish to ensure a given version of setuptools is
present.

Signed-off-by: Stephen Finucane <stephen@that.guru>

anthrotype commented:
if you add setuptools_scm to the build requirements in pyproject.toml, then you could do away with setup_requires and eggs altogether, no?

anthrotype commented:
maybe for the time being would be better to recommend to add setuptools_scm to both pyproject.toml and the setup_requires, so that newer pips will use the former (and skip the latter as already resolved by the time setup_requires are evaluated), and older pips that don't know about pyproject.toml can keep using setup_requires (provided they don't fail with some obscure SSL error).

stephenfin committed:
README: Add 'setuptools_scm' to 'pyproject.toml'

Ensure 'setuptools_scm' is defined in both 'pyproject.toml' and
'setup.cfg'. Newer pips will use the former (and skip the latter as
already resolved by the time setup_requires are evaluated), and older
pips that don't know about 'pyproject.toml' can keep using
'setup_requires'.

Signed-off-by: Stephen Finucane <stephen@that.guru>

stephenfin commented:
> maybe for the time being would be better to recommend to add setuptools_scm to both pyproject.toml and the setup_requires, so that newer pips will use the former (and skip the latter as already resolved by the time setup_requires are evaluated), and older pips that don't know about pyproject.toml can keep using setup_requires (provided they don't fail with some obscure SSL error).

Done

RonnyPfannschmidt reviewed (approved):
great work, i'm going to merge later this evening, please give me a note if you wish to streamline history a bit

stephenfin commented:
Yup, happy for you to squash the last three commits. I just kept them separate because I've had people complain about force pushing on GitHub.

RonnyPfannschmidt commented:
@stephenfin my personal stance is

* i keep commits separate while the pr is being developed
* i will squash my own commits after the review process is done and i never squash commits of others
* i give contributors the opportunity to squash themselves before i merge

stephenfin commented:
Oh, that all makes sense. In this particular case though, I'm away from a computer and won't be near one again until next week. As such, if you're happy to squash them, go for it.

marcelm commented on README.rst:
```diff
@@ -1,61 +1,119 @@
+.. important::
+
+    Ensure neither the ``[metadata] version`` option nor the ``[egg_info]``
+    section are not defined, as these will interfere with ``setuptools_scm``.
```
The extra “not” in this line should probably be removed

marcelm commented:
This PR would close #315

RonnyPfannschmidt commented:
good reminder, it got lost over the holiday season, i'll put it in my week todo
